src/benchmark.zig

```diff
@@ -72,35 +71,118 @@ pub fn main() !void {
     const cluster_id: u32 = 0;
     var address = [_]std.net.Address{try std.net.Address.parseIp4("127.0.0.1", constants.port)};
 
-    var io = try IO.init(32, 0);
-    defer io.deinit();
+    var benchmark: Benchmark = undefined;
 
-    var message_pool = try MessagePool.init(allocator, .client);
-    defer message_pool.deinit(allocator);
+    benchmark.io = try IO.init(32, 0);
 
-    var client = try Client.init(
+    benchmark.message_pool = try MessagePool.init(allocator, .client);
+
+    benchmark.client = try Client.init(
         allocator,
         client_id,
         cluster_id,
         @intCast(u8, address.len),
-        &message_pool,
+        &benchmark.message_pool,
         .{
             .configuration = address[0..],
-            .io = &io,
+            .io = &benchmark.io,
         },
     );
-    defer client.deinit(allocator);
 
-    // Init accounts.
-    var batch_accounts = try std.ArrayList(tb.Account).initCapacity(allocator, account_count_per_batch);
-    var account_index: usize = 0;
-    while (account_index < account_count) {
+    benchmark = .{
+        .io = benchmark.io,
```

Comment:
It doesn't feel right to initialize `benchmark = undefined` and then reinitialize it assigning its fields to itself, e.g. `.io = benchmark.io`.

Why not declare `io: *IO` and pass it by reference, like it is in the client initialization? Or maybe change it to initialize just once, letting `.client = undefined` to be assigned later.